### PR TITLE
Add test module to disable screensaver

### PR DESCRIPTION
**Pull request by asdil12**
Ticket: https://progress.opensuse.org/issues/40490
Needles: https://github.com/os-autoinst/os-autoinst-needles-opensuse/pull/587
Verification:
gnome: http://artemis.suse.de/tests/1660
kde: http://artemis.suse.de/tests/1659
xfce: http://artemis.suse.de/tests/1658

### CODE REVIEW

**Comment by SergioAtSUSE** [comment]
@okurz, @foursixnine, I would like you to review this in case I am missing something, since this PR is changing the schedule.

@asdil12, could you also provide a verification run for default scenario (not create_hdd)?

File: lib/utils.pm [hunk 1]
```diff
@@ -1153,6 +1153,9 @@ sub disable_serial_getty {
     # Mask if is qemu backend as use serial in remote installations e.g. during reboot
     systemctl "mask $service_name", ignore_failure => 1 if check_var('BACKEND', 'qemu');
     record_info 'serial-getty', "Serial getty mask for $testapi::serialdev";
+    # agetty changed the group from "dialout" to "tty". This won't happen again on next boot,
+    # but we need to manually change it back for now.
+    assert_script_run "chown :dialout /dev/$testapi::serialdev";
```
Comment (foursixnine) [inline lib/utils.pm]: Why?

File: lib/x11utils.pm [hunk 2]
```diff
@@ -237,4 +247,67 @@ sub turn_off_gnome_suspend {
...
+
+This function must be called from an xterm - B<not> from a bare tty.
+This is because the tty will not have the permissions to run C<xset>.
+
+=cut
+sub turn_off_x11_dpms {
+    script_run 'xset -dpms';
+    script_run 'xset s off';
+    become_root;
+    script_run "echo 'Section \"Extensions\"' > /etc/X11/xorg.conf.d/90-nodpms.conf";
```
Comment (foursixnine) [inline lib/x11utils.pm]: isn't opensuse using wayland as default now?

**Comment by asdil12** [comment]
http://artemis.suse.de/tests/1632

File: tests/x11/disable_screensaver.pm [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+use x11utils;
+
+sub run {
+    my ($self) = @_;
+
+    select_console 'x11';
+    ensure_unlocked_desktop;
+    assert_screen "generic-desktop";
+    # wait for screensaver to start
+    assert_screen [qw(screenlock screenlock-password)], 700;
```
Comment (foursixnine) [inline tests/x11/disable_screensaver.pm]: I would reduce the default timeout to 10 seconds (That way you test that the default can be changed), to not dramatically increase the duration of tests...

File: tests/x11/disable_screensaver.pm [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+    ensure_unlocked_desktop;
+
+    mouse_hide(1);
+    turn_off_screensaver;
+
+    # TODO: remove this
+    # ensure that the screensaver really got disabled
+    sleep 700;
```
Comment (foursixnine) [inline tests/x11/disable_screensaver.pm]: Same here.

File: lib/x11utils.pm [hunk 5]
```diff
@@ -237,4 +247,67 @@ sub turn_off_gnome_suspend {
...
+eg. blanking the screen and sending the monitor to sleep after some time.
+
+This will disable the setting at runtime and also add a config file for the
+XServer to make this change persist reboots.
+
+This function must be called from an xterm - B<not> from a bare tty.
+This is because the tty will not have the permissions to run C<xset>.
+
+=cut
+sub turn_off_x11_dpms {
```
Comment (foursixnine) [inline lib/x11utils.pm]: Something similar could be applied to ttys? `consoleblank=0` via grub?

File: lib/main_common.pm [hunk 6]
```diff
@@ -1098,13 +1098,11 @@ sub load_consoletests {
         loadtest "console/supportutils";
     }
     loadtest "console/check_package_version" if check_var('ORIGINAL_TARGET_VERSION', '12-SP5');
-    loadtest "console/force_scheduled_tasks" unless is_jeos;
```
Comment (foursixnine) [inline lib/main_common.pm]: This change looks unrelated

File: lib/main_common.pm [hunk 7]
```diff
@@ -1098,13 +1098,11 @@ sub load_consoletests {
...
     if (get_var("LOCK_PACKAGE")) {
         loadtest "console/check_locked_package";
     }
     loadtest "console/textinfo";
     loadtest "console/rmt" if is_rmt;
-    loadtest "console/hostname" unless is_bridged_networking;
```
Comment (foursixnine) [inline lib/main_common.pm]: This looks unrelated

Comment (asdil12) [reply lib/main_common.pm]: It is related as `hostname` is already scheduled from `load_system_prepare_tests()` function and as I added the `load_system_prepare_tests();` function to `load_reboot_tests` the `hostname` module would be scheduled twice.

File: lib/x11utils.pm (hunk 2 above)
Comment (asdil12) [reply lib/x11utils.pm]: Not for kde and xfce.

File: tests/x11/disable_screensaver.pm (hunk 4 above)
Comment (asdil12) [reply tests/x11/disable_screensaver.pm]: When reconfiguration wouldn't have worked, I wouldn't detect it then.

File: lib/main_common.pm (hunk 6 above)
Comment (asdil12) [reply lib/main_common.pm]: see my comment regarding hostname.

File: tests/x11/disable_screensaver.pm (hunk 4 above)
Comment (foursixnine) [reply tests/x11/disable_screensaver.pm]: So reconfiguring both to say, 10-60 seconds should allow you to detect the case when it didn't work.

File: tests/x11/disable_screensaver.pm (hunk 3 above)
Comment (asdil12) [reply tests/x11/disable_screensaver.pm]: reduced to 1m

File: tests/x11/disable_screensaver.pm (hunk 4 above)
Comment (asdil12) [reply tests/x11/disable_screensaver.pm]: 1m

File: lib/x11utils.pm (hunk 5 above)
Comment (asdil12) [reply lib/x11utils.pm]: I don't think this is needed as we already run `$self->script_run('setterm -blank 0')` from `lib/susedistribution.pm` in the `activate_console` method.
